demo/web/firebase_element/x_login.dart

```diff
@@ -0,0 +1,112 @@
+@HtmlImport('x_login.html')
+library x_login;
+
+import 'dart:html' as dom;
+import 'dart:convert' show JSON;
+import 'package:web_components/web_components.dart' show HtmlImport;
+import 'package:polymer/polymer.dart';
+import 'package:polymer_elements/firebase_auth.dart';
+
+/// Silence analyzer [FirebaseAuth]
+@PolymerRegister('x-login')
+class XLogin extends PolymerElement {
+  XLogin.created() : super.created();
+
+  @property String provider = 'anonymous';
+  @property String message = '';
+  @property String email = '';
+  @property String password = '';
+  @property String newPassword = ''; //
+  @property Map user;
+  @property bool statusKnown = false;
+  @property String params; //
+
+  @eventHandler
+  void login([_, __]) {
+    String params;
+    Map parameters;
+    try {
+      parameters = JSON.decode(params);
+    } catch (e) {
+      parameters = null;
+    }
+
+    if (provider == 'password') {
+      parameters = parameters ?? {};
+      parameters['email'] = email;
+      parameters['password'] = password;
+    }
+
+    _firebaseLogin.login(parameters, null);
+  }
+
+  FirebaseAuth get _firebaseLogin => $['firebaseLogin'];
+
+  @eventHandler
+  void logout([_, __]) => _firebaseLogin.logout();
+
+  @eventHandler
+  errorHandler(dom.Event e, var detail) =>
+      message = 'Error: ${detail['message']}';
```

Comment:
This should be `set(message, 'Error: ${detail['message']}')`, or call notifyPath. You could also define a getter/setter for the property (and call notifyPath in the setter).